src/main/kotlin/com/terraformation/backend/customer/FacilityService.kt

```diff
@@ -41,6 +45,42 @@ class FacilityService(
         .forEach { facility -> publisher.publishEvent(FacilityTimeZoneChangedEvent(facility)) }
   }
 
+  @EventListener
+  fun on(event: ReportSubmittedEvent) {
+    event.body.nurseries.forEach { nurseryBody ->
+      val existing = facilityStore.fetchOneById(nurseryBody.id)
+      val updated =
+          existing.copy(
+              buildCompletedDate = existing.buildCompletedDate ?: nurseryBody.buildCompletedDate,
+              buildStartedDate = existing.buildStartedDate ?: nurseryBody.buildStartedDate,
+              capacity = existing.capacity ?: nurseryBody.capacity,
+              operationStartedDate = existing.operationStartedDate
+                      ?: nurseryBody.operationStartedDate,
```

Comment:
If a report changes these values (or the seedbank ones) after they have been set once, they won't be updated here. But they "new values" will still go out with the report. Is there no way to update these and is that intended?

Reply:
The flow here is a little weird because there's a bidirectional data dependency, but what would happen in that case would be that the fields wouldn't be considered editable any more, and the new values from the subsequent report would be overwritten by `ReportService.fetchOneById()` during the PUT request.

The idea (or rather, the requirement per the PRD) is that submitting a report can only set initial values for these fields on the facility. You can use the facility edit UI to update them afterwards, but once they're set, they're considered read-only in the report data.

Reply:
Ah that makes sense. Thank you.